### PR TITLE
Revise args of `$`

### PR DESCRIPTION
**Pull request by Unit112358**
please check whether it is correct.

### CODE REVIEW

File: src/il/tos.nim [hunk 1]
```diff
@@ -47,111 +47,158 @@ proc `$`*(self: Literal): string =
...
+            rety = `$`(self.rety, typed, regioned, comment)
+            args = self.args.mapIt(`$`(it, typed, regioned, comment)).join(", ")
+        fmt"func({args}) -> {rety}"
     of ExpressionKind.IntCast:
+        let
+            int_exp = `$`(self.int_exp, typed, regioned, comment)
         fmt"cast({self.int_exp}, {self.from}, {self.to})"
     of ExpressionKind.Fail:
         fmt"failed term"
     if not self.typ.isNil:
```
Comment (chocobo333) [inline src/il/tos.nim:192]:
```suggestion
if typed and not self.typ.isNil:
```

File: src/il/tos.nim [hunk 2]
```diff
@@ -162,156 +209,184 @@ proc `$`*(self: Metadata): string =
...
     of StatementKind.Discard:
         if self.`discard`.isSome:
-            fmt"discard {self.`discard`.get}"
+            let dscrd = `$`(self.`discard`.get, typed, regioned, comment)
+            fmt"discard {dscrd}"
         else:
             fmt"discard"
     of StatementKind.Comments:
-        self.comments.join("\n")
+        self.comments.mapIt(`$`(it, typed, regioned, comment)).join("\n")
```
Comment (chocobo333) [inline src/il/tos.nim]:
```suggestion
if comment:
    self.comments.mapIt(`$`(it, typed, regioned, comment)).join("\n")
else:
    ""
```

**Review by chocobo333 — CHANGES_REQUESTED** [review]

File: src/il/tos.nim (hunk 1 above)
Comment (Unit112358) [reply src/il/tos.nim:192]: Revised

File: src/il/tos.nim (hunk 2 above)
Comment (Unit112358) [reply src/il/tos.nim]: Revised

**Review by chocobo333 — APPROVED** [review]

**Pull request merged**